cypress/support/commands.js

```diff
@@ -0,0 +1,40 @@
+// ***********************************************
+// This example commands.js shows you how to
+// create various custom commands and overwrite
+// existing commands.
+//
+// For more comprehensive examples of custom
+// commands please read more here:
+// https://on.cypress.io/custom-commands
+// ***********************************************
+//
+//
+// -- This is a parent command --
+// Cypress.Commands.add('login', (email, password) => { ... })
+//
+//
+// -- This is a child command --
+// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
+//
+//
+// -- This is a dual command --
+// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+
+const url = Cypress.config('baseUrl').replace(/\/index.php\/?$/g, '')
+Cypress.env('baseUrl', url)
+
+Cypress.Commands.add('login', (user, password, route = '/apps/deck/') => {
+	cy.session(user, function () {
+		cy.visit(route)
+		cy.get('input[name=user]').type(user)
+		cy.get('input[name=password]').type(password)
+		cy.get('#submit-wrapper input[type=submit]').click()
+		cy.url().should('include', route)
+	})
+	// in case the session already existed but we are on a different route...
+	cy.visit(route)
+})
```

Comment:
Maybe we add https://github.com/nextcloud/text/blob/master/cypress/support/commands.js#L41-L96 also right away, then the basic test could already use a random user as in https://github.com/nextcloud/text/blob/master/cypress/integration/share.spec.js#L25-L26